#### Getting to know the data:
venv_scientific - The virtualenv with pandas, numpy, scipy, matplotlib, 

In [2]:
import pandas as pd

In [3]:
from subprocess import check_output
print(check_output(["ls", "data"]).decode("utf8"))

samplesubmission.csv
test.csv
train.csv



train.csv, test.csv and sample_submission.csv The train data consists of sample projects from the May 2009 to May 2015. The test data consists of projects from June 2015 to March 2017.

In [4]:
train_data = pd.read_csv("data/train.csv")
test_data = pd.read_csv("data/test.csv")

In [5]:
train_data.shape

(108129, 14)

In [6]:
test_data.shape

(63465, 12)

In [18]:
train_data.head()

project_id 
 name 
 desc 
 goal 
 keywords 
 disable_communication 
 country 
 currency 
 deadline 
 state_changed_at 
 created_at 
 launched_at 
 backers_count 
 final_status 
 
 
 
 
 0 
 kkst1451568084 
 drawing for dollars 
 I like drawing pictures. and then i color them too. so i thought i would suggest something for m... 
 20.0 
 drawing-for-dollars 
 False 
 US 
 USD 
 1241333999 
 1241334017 
 1240600507 
 1240602723 
 3 
 1 
 
 
 1 
 kkst1474482071 
 Sponsor Dereck Blackburn (Lostwars) Artist in Residence in Kankakee Illinois 
 I, Dereck Blackburn will be taking upon an incredible journey in the month of May 2009. I will b... 
 300.0 
 sponsor-dereck-blackburn-lostwars-artist-in-residence-in-kankakee-illinois 
 False 
 US 
 USD 
 1242429000 
 1242432018 
 1240960224 
 1240975592 
 2 
 0 
 
 
 2 
 kkst183622197 
 Mr. Squiggles 
 So I saw darkpony's successfully funded drawing for dollars project and I thought """"""""""""""... 
 30.0 
 mr-squiggles 
 False 
 US 
 USD 
 1243027560 
 1243027818 
 1242163613 
 1242164398 
 0 
 0 
 
 
 3 
 kkst597742710 
 Help me write my second novel. 
 Do your part to help out starving artists and help me write my second novel!\r\n\r\nI have just ... 
 500.0 
 help-me-write-my-second-novel 
 False 
 US 
 USD 
 1243555740 
 1243556121 
 1240963795 
 1240966730 
 18 
 1 
 
 
 4 
 kkst1913131122 
 Support casting my sculpture in bronze 
 I'm nearing completion on a sculpture, currently titled """"""""""""""""""""""""""""""""""""""""... 
 2000.0 
 support-casting-my-sculpture-in-bronze 
 False 
 US 
 USD 
 1243769880 
 1243770317 
 1241177914 
 1241180541 
 1 
 0

In [35]:
test_data.head(4)

project_id                                               name  \
0   kkst917493670                                          Bràthair.   
1  kkst1664901914                                   THE SCREENWRITER   
2   kkst925125077        The Hornets Nest the Fairmont Heights Story   
3  kkst1427645275  BROTHERS Season 2 - Groundbreaking Transgender...   

                                                desc     goal  \
0  My first film, of many to come. Trying to purs...   7000.0   
1  A young man that has earned his master's in sc...  35000.0   
2  Film about a high school constructed for negro...  49500.0   
3  The acclaimed series about a group of transgen...  40000.0   

                                            keywords  disable_communication  \
0                                           brathair                  False   
1                                   the-screenwriter                  False   
2        the-hornets-nest-the-fairmont-heights-story                  False   
3  brothers-season-2-groundbreaking-transgender-male                  False   

  country currency    deadline  state_changed_at  created_at  launched_at  
0      US      USD  1449619185        1449619185  1446002581   1446159585  
1      US      USD  1453435620        1453435620  1450297323   1450411620  
2      US      USD  1451780700        1451780700  1448581356   1448672128  
3      US      USD  1445021518        1445021530  1440966830   1442429518

# final_status:
whether the project was successful or not 
1 -> success
0 -> failed

In [36]:
print(train_data['final_status'].value_counts())


0    73568
1    34561
Name: final_status, dtype: int64


In [39]:
100 * train_data.final_status.value_counts()/len(train_data.final_status)

0    68.037252
1    31.962748
Name: final_status, dtype: float64

# goal - 
the goal (amount) required for the project.
0.01 to 100,000,000 asked.
Mean - $ 36726

Successful projects mean - below 10,000 - <b>9806</b>
Failed projects mean - is nearly 50,000 - <b>49372.5889115</b>

In [17]:
pd.set_option('display.max_colwidth', 100)

In [18]:
print(train_data['goal'].idxmax())  # index of max
print(train_data['goal'].max()) # max value for the column
train_data.index  # data index


65916
100000000.0


RangeIndex(start=0, stop=108129, step=1)

In [13]:
print(train_data['goal'].idxmin())  # index of min
print(train_data['goal'].min())

317
0.01


In [26]:
train_data.index.tolist()  # get the DataFrame index as a list.


array([     0,      1,      2, ..., 108126, 108127, 108128])

In [27]:
train_data.index.get_values()  # a better and new approach


array([     0,      1,      2, ..., 108126, 108127, 108128])

In [28]:
print(train_data['goal'].mean())  # average/mean of the column

36726.2288326


In [40]:
print(train_data['goal'][train_data['final_status'] == 1].mean())  # average/mean of successful projects

9806.65421718


In [41]:
print(train_data['goal'][train_data['final_status'] == 0].mean())  # average/mean of failed projects

49372.5889115


# disable_communication - 
whether the project authors has disabled communication option with people donating to the project

In [33]:
print(train_data['disable_communication'].value_counts())


False    107806
True        323
Name: disable_communication, dtype: int64


In [45]:
print(100 * train_data['disable_communication'].value_counts()) / len(train_data.disable_communication)


False    99.701283
True      0.298717
Name: disable_communication, dtype: float64


In [34]:
print(train_data['disable_communication'][train_data['final_status'] == 1].value_counts())


False    34561
Name: disable_communication, dtype: int64


In [43]:
print(train_data['disable_communication'][train_data.final_status == 0].value_counts())


False    73245
True       323
Name: disable_communication, dtype: int64


A rather insignificant feature - as <b>99.7%</b> have <b>it False</b>

# duration:
A new feature, stating project duration in terms of days it was opne/active

In [51]:
# processing the UNIX timestamps into readable dates:
cols = ['state_changed_at','created_at','launched_at','deadline']
for col in cols:
    train_data['%s' % col] = pd.to_datetime(train_data['%s' % col], unit='s')

In [52]:
train_data.head(2)

project_id                                               name  \
0  kkst1451568084                                drawing for dollars   
1  kkst1474482071  Sponsor Dereck Blackburn (Lostwars) Artist in ...   

                                                desc   goal  \
0  I like drawing pictures. and then i color them...   20.0   
1  I, Dereck Blackburn will be taking upon an inc...  300.0   

                                            keywords  disable_communication  \
0                                drawing-for-dollars                  False   
1  sponsor-dereck-blackburn-lostwars-artist-in-re...                  False   

  country currency            deadline    state_changed_at  \
0      US      USD 2009-05-03 06:59:59 2009-05-03 07:00:17   
1      US      USD 2009-05-15 23:10:00 2009-05-16 00:00:18   

           created_at         launched_at  backers_count  final_status  
0 2009-04-24 19:15:07 2009-04-24 19:52:03              3             1  
1 2009-04-28 23:10:24 2009-04-29 03:26:32              2             0

In [59]:
# dt - only work for datetime series - TypeError if the Series does not contain datetimelike values.
train_data['project_duration'] = (train_data.deadline - train_data.launched_at).dt.days

In [57]:
train_data.head(2)

project_id                                               name  \
0  kkst1451568084                                drawing for dollars   
1  kkst1474482071  Sponsor Dereck Blackburn (Lostwars) Artist in ...   

                                                desc   goal  \
0  I like drawing pictures. and then i color them...   20.0   
1  I, Dereck Blackburn will be taking upon an inc...  300.0   

                                            keywords  disable_communication  \
0                                drawing-for-dollars                  False   
1  sponsor-dereck-blackburn-lostwars-artist-in-re...                  False   

  country currency            deadline    state_changed_at  \
0      US      USD 2009-05-03 06:59:59 2009-05-03 07:00:17   
1      US      USD 2009-05-15 23:10:00 2009-05-16 00:00:18   

           created_at         launched_at  backers_count  final_status  \
0 2009-04-24 19:15:07 2009-04-24 19:52:03              3             1   
1 2009-04-28 23:10:24 2009-04-29 03:26:32              2             0   

   project_duration  
0                 8  
1                16

In [62]:
train_data.project_duration.value_counts()



30    44506
29     6787
60     6736
45     4421
31     2943
59     2615
40     2573
35     2513
20     2048
21     2009
25     1757
28     1753
32     1743
33     1499
15     1450
14     1384
44     1161
34     1153
27     1023
24      851
39      830
36      812
26      763
22      739
10      664
37      660
23      657
50      641
38      598
42      543
      ...  
88       65
70       54
63       53
62       51
65       49
75       47
66       43
1        37
80       36
2        36
86       34
91       32
72       32
71       32
87       29
76       27
85       25
84       24
68       22
77       22
69       22
83       20
67       20
78       19
81       19
64       18
79       17
82       16
73       15
74       13
Name: project_duration, Length: 91, dtype: int64

In [65]:
train_data.project_duration.mean()



33.917034283124785

In [64]:
train_data.project_duration[train_data.final_status == 1].mean()



32.124475564943147

In [66]:
train_data.project_duration[train_data.final_status == 0].mean()


34.759147999130057

Not much si